features/wallet/impl/src/main/java/com/tangem/feature/wallet/presentation/wallet/state/factory/TokenActionsProvider.kt

```diff
@@ -1,48 +1,75 @@
 package com.tangem.feature.wallet.presentation.wallet.state.factory
 
-import com.tangem.domain.tokens.model.CryptoCurrencyStatus
+import com.tangem.core.ui.extensions.TextReference
+import com.tangem.domain.tokens.model.TokenActionsState
 import com.tangem.feature.wallet.impl.R
 import com.tangem.feature.wallet.presentation.common.state.TokenItemState
 import com.tangem.feature.wallet.presentation.wallet.state.TokenActionButtonConfig
 import com.tangem.feature.wallet.presentation.wallet.viewmodels.WalletClickIntents
 import kotlinx.collections.immutable.ImmutableList
 import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.transform
 
 /**
  * Converter from loaded [TokenItemState.Content] to ImmutableList<[TokenActionButtonConfig]>
  *
  * @property clickIntents screen click intents
  *
  */
+@Suppress("UnusedPrivateMember") // will be used in next PRs
 internal class TokenActionsProvider(private val clickIntents: WalletClickIntents) {
 
-    fun provideActions(cryptoCurrencyStatus: CryptoCurrencyStatus): ImmutableList<TokenActionButtonConfig> {
-        // TODO: https://tangem.atlassian.net/browse/AND-4204
-        return mockTokenActionButtonConfig(cryptoCurrencyStatus).toImmutableList()
+    fun provideActions(tokenActions: Flow<TokenActionsState>): Flow<ImmutableList<TokenActionButtonConfig>> {
+        return tokenActions.transform {
+            emit(convertTokenActionsState(it))
+        }
     }
 
-    private fun mockTokenActionButtonConfig(cryptoCurrencyStatus: CryptoCurrencyStatus): List<TokenActionButtonConfig> {
-        return listOf(
-            TokenActionButtonConfig(
-                text = "Send",
-                iconResId = R.drawable.ic_plus_24,
-                onClick = { clickIntents.onMultiCurrencySendClick(cryptoCurrencyStatus) },
-            ),
-            TokenActionButtonConfig(
-                text = "Buy",
-                iconResId = R.drawable.ic_plus_24,
-                onClick = {},
-            ),
-            TokenActionButtonConfig(
-                text = "Sell",
-                iconResId = R.drawable.ic_plus_24,
-                onClick = {},
-            ),
-            TokenActionButtonConfig(
-                text = "Swap",
-                iconResId = R.drawable.ic_plus_24,
-                onClick = {},
-            ),
+    private fun convertTokenActionsState(
+        tokenActionsState: TokenActionsState,
+    ): ImmutableList<TokenActionButtonConfig> {
+        return tokenActionsState.states
+            .map(::tokenActionStateMapper)
+            .toImmutableList()
+    }
+
+    private fun tokenActionStateMapper(actionsState: TokenActionsState.ActionState): TokenActionButtonConfig {
+        val title: TextReference
+        val icon: Int
+        val action: () -> Unit
+        when (actionsState) {
+            is TokenActionsState.ActionState.Buy -> {
+                title = TextReference.Res(R.string.common_buy)
```

Comment:
Я создал методы для референсов, думаю стоит их заюзать
```suggestion
                title = resourceReference(R.string.common_buy)
```

Reply:
добавил